io/src/real_sense_2_grabber.cpp

```diff
@@ -0,0 +1,219 @@
+#include <pcl/io/eigen.h>
+#include <pcl/io/eigen.h>
+#include <pcl/io/boost.h>
+#include <pcl/io/grabber.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+
+#include <librealsense2/rs.hpp>
+#include <pcl/io/real_sense_2_grabber.h>
+
+namespace pcl
+{
+
+RealSense2Grabber::RealSense2Grabber()
+: running(false)
+, quit(false)
+, signal_PointXYZ(nullptr)
+, signal_PointXYZI(nullptr)
+, signal_PointXYZRGB(nullptr)
+, signal_PointXYZRGBA(nullptr)
+{
+    
+    // for now we only create the xyzrgb signal
+    signal_PointXYZ = createSignal<signal_librealsense_PointXYZ>();
+    signal_PointXYZI = createSignal<signal_librealsense_PointXYZI>();
+    signal_PointXYZRGB = createSignal<signal_librealsense_PointXYZRGB>();
+    signal_PointXYZRGBA = createSignal<signal_librealsense_PointXYZRGBA>();
+}
+
+
+RealSense2Grabber::~RealSense2Grabber()
+{
+    stop();
+
+    disconnect_all_slots<signal_librealsense_PointXYZ>();
+    disconnect_all_slots<signal_librealsense_PointXYZI>();
+    disconnect_all_slots<signal_librealsense_PointXYZRGB>();
+    disconnect_all_slots<signal_librealsense_PointXYZRGBA>();
+
+    thread.join();
+
+    pipe.stop();    
+}
+
+void RealSense2Grabber::start()
+{
+    running = true;
+
+	rs2::config cfg;
+
+	cfg.enable_stream(RS2_STREAM_COLOR, 424, 240, RS2_FORMAT_RGB8, 30);
+	cfg.enable_stream(RS2_STREAM_DEPTH, 424, 240, RS2_FORMAT_ANY, 30);
+	cfg.enable_stream(RS2_STREAM_INFRARED, 424, 240, RS2_FORMAT_ANY, 30);
+
+	pipe.start(cfg);
+
+    thread = boost::thread(&RealSense2Grabber::threadFunction, this);
+}
+
+void RealSense2Grabber::stop()
+{
+    boost::unique_lock<boost::mutex> lock(mutex);
+
+    quit = true;
+    running = false;
+
+    lock.unlock();
+}
+
+bool RealSense2Grabber::isRunning() const
+{
+    boost::unique_lock<boost::mutex> lock(mutex);
+
+    return running;
+
+    lock.unlock();
+}
+
+float RealSense2Grabber::getFramesPerSecond() const
+{
+    return 30.0f;
+}
+
+pcl::PointCloud<pcl::PointXYZ>::Ptr RealSense2Grabber::convertDepthToPointXYZ(const rs2::points & points)
+{
+    return pcl::PointCloud<pcl::PointXYZ>::Ptr();
+}
+
+pcl::PointCloud<pcl::PointXYZI>::Ptr RealSense2Grabber::convertInfraredDepthToPointXYZI(const rs2::points & points)
+{
+    return pcl::PointCloud<pcl::PointXYZI>::Ptr();
+}
+
+pcl::PointCloud<pcl::PointXYZRGB>::Ptr RealSense2Grabber::convertRGBDepthToPointXYZRGB(const rs2::points & points, rs2::video_frame & rgb)
+{
+    pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZRGB>());
+
+    auto sp = points.get_profile().as<rs2::video_stream_profile>();
+    cloud->width = sp.width();
+    cloud->height = sp.height();
+    cloud->is_dense = false;
+    cloud->points.resize(points.size());
+
+    auto ptr = points.get_vertices();
+    auto uvptr = points.get_texture_coordinates();
+
+    uint8_t r, g, b;
+    for (auto& p : cloud->points)
```

Comment:
Please consider to parallelizing using OpenMP.

Reply:
I've added OpenMP to the loops.  Please let me know if I'm using the standard PCL OpenMP conventions.